cpp/src/parquet/encoding.cc

```diff
@@ -2183,15 +2185,21 @@ class DeltaBitPackEncoder : public EncoderImpl, virtual public TypedEncoder<DTyp
 
   void FlushBlock();
 
+ private:
+  constexpr T SafeSubtract(T a, T b) const {
+    constexpr WT mask = static_cast<WT>(static_cast<UT>(static_cast<T>(-1)));
+    return static_cast<T>((static_cast<WT>(a) - static_cast<WT>(b)) & mask);
+  }
```

Comment:
Would this maybe better fit in [int_util_overflow.h](https://github.com/apache/arrow/blob/main/cpp/src/arrow/util/int_util_overflow.h)?

Reply:
Yes, that sounds like a better location. Thanks!